src/js/wrapper.ts

```diff
@@ -395,10 +397,25 @@ export const NATIVE: SentryNativeWrapper = {
       throw this._NativeClientError;
     }
 
-    // we stringify the extra as native only takes in strings.
-    const stringifiedExtra = typeof extra === 'string' ? extra : JSON.stringify(extra);
+    if (typeof extra === 'string') {
+      return RNSentry.setExtra(key, extra);
+    }
+    if (typeof extra === 'undefined') {
+      return RNSentry.setExtra(key, 'undefined');
+    }
+
+    let stringifiedExtra: string | undefined;
+    try {
+      const normalizedExtra = normalize(extra);
+      stringifiedExtra = JSON.stringify(normalizedExtra);
+    } catch (e) {
+      logger.error('Extra not passed to native SDK, because it contains non-stringifiable values', e);
```

Comment:
l: We could add the `key` to the error for easier debugging.
```suggestion
      logger.error('Extra for key ${key} not passed to native SDK, because it contains non-stringifiable values', e);
```